### PR TITLE
[datesets] feat: add statsd to datasets api

### PR DESCRIPTION
**Pull request by lilykuang**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Adding statds metrics to datesets API endpoints (POST, PUT, DELETE, bulk_delete, data). This is an extension from #9519
### TEST PLAN

- Deploy this PR to sandbox instance
- Hit different points by creating a new chart, updating chart owners and deleting charts, etc
- Go to https://app.datadoghq.com/metric/explorer
- Check on graphs for `superset.DatasetRestApi.[endpoints]` over `environment:sandbox`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/datasets/api.py [hunk 1]
```diff
@@ -344,6 +352,7 @@ def export(self, **kwargs: Any) -> Response:
 
     @expose("/<pk>/refresh", methods=["PUT"])
     @protect()
+    @statsd_metrics
```
Comment (dpgaspar) [inline superset/datasets/api.py:355]: Change the position, after the `safe` decorator

**Review by dpgaspar — COMMENTED** [review]
Looks good, just a comment

**Comment by dpgaspar** [comment]
Can you please fill the "Summary" to add a bit of context to this PR?

File: tests/datasets/api_tests.py [hunk 2]
```diff
@@ -662,7 +662,7 @@ def test_dataset_item_refresh(self):
 
     def test_dataset_item_refresh_not_found(self):
         """
-            Dataset API: Test item refresh not found dataset
+        Dataset API: Test item refresh not found dataset
```
Comment (dpgaspar) [inline tests/datasets/api_tests.py:665]: Can we add the metric assert here?

File: tests/datasets/api_tests.py [hunk 3]
```diff
@@ -673,7 +673,7 @@ def test_dataset_item_refresh_not_found(self):
 
     def test_dataset_item_refresh_not_owned(self):
         """
-            Dataset API: Test item refresh not owned dataset
+        Dataset API: Test item refresh not owned dataset
```
Comment (dpgaspar) [inline tests/datasets/api_tests.py:676]: Can we add the metric assert here?

**Review by dpgaspar — CHANGES_REQUESTED** [review]
Just missing 2 metric assertions

**Review by etr2460 — COMMENTED** [review]
Could you add a test plan to the PR as well? Thanks!

**Review by etr2460 — APPROVED** [review]
cool, makes sense to me. thanks for the test plan!

**Review by dpgaspar — APPROVED** [review]
Nice!

**Pull request merged**